httputil/retry.go

```diff
@@ -139,6 +193,10 @@ func RetryRequest(endpoint string, doRequest func() (*http.Response, error), rea
 			if ShouldRetryError(attempt, err) {
 				continue
 			}
+
+			if isNetworkDown(err) || isDnsUnavailable(err) {
+				err = &UnretriedNetworkError{Err: err}
```

Comment:
so if I understand correctly the isDnsUnavailble case is retried but we want to retry further if we didn't succeed after the given retries?

Reply:
Yes. We lie a bit by saying UnretriedNetworkError; only the case on 16.04 though.

Reply:
Ok, maybe PerstistentNetworkError would be a better name though

Reply:
Renamed.